app/models/submission.rb

```diff
@@ -54,19 +54,60 @@ def broadcast!
   #   end
   # end
 
+  def automated_feedback_status
+    # If they're both still waiting, then return pending
+    return :pending if !representation_generated? && !analysis_completed?
+
+    # If either has feedback then we're present
+    return :present if exercise_representation&.has_feedback? || analysis&.has_feedback?
+
+    # Otherwise if either are still queued then we're pending
+    return :pending if representation_queued? || representation_not_queued? ||
+                       analysis_queued? || analysis_not_queued?
+
+    # Otherwise we don't have feedback
+    :none
+  end
+
   memoize
   def has_automated_feedback?
-    exercise_representation&.has_feedback? || analysis&.has_feedback?
+    automated_feedback_status == :present
   end
 
   memoize
   def automated_feedback
-    feedback = []
-    feedback << exercise_representation if exercise_representation&.has_feedback?
-    feedback
+    return nil unless has_automated_feedback?
+
+    exercise_representation&.has_feedback? ? representer_feedback : analyzer_feedback
```

Comment:
This means that can have either representer feedback or analyzer feedback, but not both, right? I think this makes sense.

Reply:
Yes, although now we're changing that!